test/test_jit.py

```diff
@@ -7892,6 +7892,17 @@ def test_as_tensor_tensor_input(input):
         inp = torch.randn(3, 4)
         self.checkScript(test_as_tensor_tensor_input, (inp,))
 
+    def test_torch_tensor_dtype(self):
+        def foo(s: float):
+            return torch.tensor(s)
+
+        scripted_foo = torch.jit.script(foo)
```

Comment:
For extra safety: wrap setting the default_dtype in a try block and restore its value in the finally clause

Reply:
is there a context manager that's already defined for this?

Reply:
You would think so, right? Want to add one?